pallets/auctions/src/lib.rs

```diff
@@ -0,0 +1,726 @@
+        /// Only owner of NFT can list for auction
+        NftNotOwned,
+        /// buy_it_price should be greater then start_price
+        AuctionPricingInvalid,
+        /// NFT has not been listed for auction
+        NFTNotListedForAuction,
+        /// NFT is already listed for sale
+        NFTAlreadyListedForSale,
+        /// The given NFTID is invalid
+        NFTIdInvalid,
+        /// The nft is in transmission
+        NFTInTransmission,
+        /// The nft is not currently listed for sale
+        NFTNotListedForSale,
+        /// The specified auction does not exist
+        AuctionDoesNotExist,
+        /// Current owner cannot bid on NFT
+        OwnerCannotCreateBid,
+        /// The auction has not started
+        AuctionNotStarted,
+        /// The auction has already ended
+        AuctionEnded,
+        /// The bid amount is lower than current highest bid
+        InvalidBidAmount,
+        /// Unexpected error occured
+        UnexpectedError,
+        /// Auction start should be greater than end block
+        AuctionStartBlockLesserThanEndBlock,
+        /// Auction start block must be higher than current block
+        AuctionStartLowerThanCurrentBlock,
+        /// Auction time should be higher than min auction duration
+        AuctionTimelineLowerThanMinDuration,
+        /// Auction time should be lower than max auction duration
+        AuctionTimelineGreaterThanMaxDuration,
+        /// The specified bid does not exist
+        BidDoesNotExist,
+        /// Cannot cancel an auction that has started
+        CannotCancelInProcessAuction,
+        /// User cannot create more than one bid
+        UserBidAlreadyExists,
+        /// The auction does not have a buy it now price
+        AuctionDoesNotSupportBuyItNow,
+        /// The auction has not yet completed
+        AuctionNotCompleted,
+    }
+
+    #[pallet::hooks]
+    impl<T: Config> Hooks<BlockNumberFor<T>> for Pallet<T> {}
+
+    #[pallet::call]
+    impl<T: Config> Pallet<T> {
+        /// An example dispatchable that takes a singles value as a parameter, writes the value to
+        /// storage and emits an event. This function must be dispatched by a signed extrinsic.
+        #[pallet::weight(10_000 + T::DbWeight::get().writes(1))]
+        #[transactional]
+        pub fn create_auction(
+            origin: OriginFor<T>,
+            nft_id: NFTId,
+            marketplace_id: MarketplaceId,
+            #[pallet::compact] start_block: T::BlockNumber,
+            #[pallet::compact] end_block: T::BlockNumber,
+            start_price: BalanceOf<T>,
+            buy_it_price: Option<BalanceOf<T>>,
+        ) -> DispatchResultWithPostInfo {
+            let creator = ensure_signed(origin)?;
+            let current_block = frame_system::Pallet::<T>::block_number();
+            // ensure the auction timeline is valid
+            ensure!(
+                start_block < end_block,
+                Error::<T>::AuctionStartBlockLesserThanEndBlock
```

Comment:
This is a great step forward with Error naming but we need to go even further. Those errors are read by our users so they need to be articulated in that way. A good name for this error would be 
`Error::<T>::AuctionCannotEndBeofreItHasStarted`

Do the same for all Errors, make them "user" readable :)